compiler/luci/lang/include/luci/IR/CircleNodeDecl.h

```diff
@@ -61,6 +62,9 @@ struct CircleNode : public loco::Node,
     _sparsityparam = std::move(sparsityparam);
   }
 
+  ShapeSignature shape_signature(void) const { return _shape_signature; }
+  void shape_signature(ShapeSignature ss) { _shape_signature = ss; }
```

Comment:
compared to `ShapeStatus`, `ShapeSignature` is a class with `std::vector` member.
These two will always make a one more more copies of this member when using these.
Is this your intention?

Reply:
I didn't thought about this point, I will fix it. Thanks!

Reply:
`void shape_signature(ShapeSignature ss) { _shape_signature = ss; }` vs
`void shape_signature(ShapeSignature &ss) { _shape_signature = ss; }` : can you describe the difference?

Reply:
It was good chance to study about it :)